docs/src/main/sphinx/routines/examples.md

```diff
@@ -369,3 +369,34 @@ BEGIN
   RETURN r;
 END
 ```
+
+## Date String Parsing example
+
+This routine parses a date string (VARCHAR) into TIMESTAMP. Date strings are commonly represented by ISO 8601 (e.g. 2023-12-01). Date strings can also be represented by the YYYYmmdd and YYYYmmddHH format (e.g. 20230101, 2023010123). For example, Hive day and hourly partitions can use these formats (e.g. '/day=20230101', '/hour=2023010123'). 
+This routine parses date strings in a best-effort fashion and can be used as a replacement for date string manipulation functions such as `date`, `date_parse`, `from_iso8601_date`,  and `from_iso8601_timestamp`.
+
+```sql
+WITH
+  FUNCTION from_date_string(date_string VARCHAR)
+  RETURNS TIMESTAMP WITH TIME ZONE
```

Comment:
Maybe talk about why you use this data type .. and where you get timezone data from?

Reply:
I have to use this data type since `from_iso8601_timestamp` function returns `timestamp(3) with time zone` 
https://trino.io/docs/current/functions/datetime.html#from_iso8601_timestamp

I'd rather use a regular `timestamp` return type instead to not deal with timezone. Any suggestions here?

Reply:
`from_iso8601_timestamp` doc mentions that 
```
The time defaults to 00:00:00.000, and the time zone defaults to the session time zone
```

I'll add this to the routine description

Reply:
But you can also leave it and call it out as a issue on the current approach